src/components/Modal/index.js

```diff
@@ -1,139 +1,72 @@
-    INSIDE: 'inside-right',
-    INSIDE_RIGHT: 'inside-right',
-    INSIDE_LEFT: 'inside-left',
-    OUTSIDE: 'outside-right',
-    OUTSIDE_RIGHT: 'outside-right',
-    OUTSIDE_LEFT: 'outside-left',
-    NONE: 'none',
-  }
 
+export default class Modal extends PureComponent {
   static propTypes = {
-    isOpen: bool.isRequired,
-    onClose: func.isRequired,
-    children: oneOfType([arrayOf(node), node]).isRequired,
-    shouldCloseOnEsc: bool,
-    shouldCloseOnClickOutside: bool,
-    className: string,
-    header: string,
-    subheader: string,
-    HeaderComponent: func,
-    closeButtonPosition: oneOf([
-      'inside-right',
-      'inside-left',
-      'outside-right',
-      'outside-left',
-      'none',
+    children: PropTypes.oneOfType([
+      PropTypes.node,
+      PropTypes.arrayOf(PropTypes.node),
     ]),
-    mountEntry: string,
-  }
-
-  static defaultProps = {
-    shouldCloseOnEsc: true,
-    shouldCloseOnClickOutside: true,
-    className: '',
-    HeaderComponent: ModalHeader,
-    closeButtonPosition: 'outside-right',
-    mountEntry: 'modal-root',
-  }
-
-  componentDidMount () {
-    document.addEventListener('keydown', this.onPressEsc)
-  }
+    className: PropTypes.string,
+    show: PropTypes.bool,
 
-  componentWillUnmount () {
-    document.removeEventListener('keydown', this.onPressEsc)
+    onCloseClick: PropTypes.func,
   }
 
-  onPressEsc = (event) => {
-    const { onClose, shouldCloseOnEsc } = this.props
-    const escKey = 27
-    const escKeyWasPressed = event.keyCode === escKey
-    if (shouldCloseOnEsc && escKeyWasPressed) {
-      onClose()
+  onKeyDown = (event) => {
+    if (event.keyCode === 27) {
+      this.close()
     }
   }
 
-  onClickOutside = () => {
-    const { onClose, shouldCloseOnClickOutside } = this.props
-    if (shouldCloseOnClickOutside) {
-      onClose()
+  close = () => {
+    const {
+      onCloseClick,
+    } = this.props
+
+    if (onCloseClick) {
+      onCloseClick()
     }
   }
 
-  container = React.createRef();
-
   render () {
     const {
-      isOpen,
-      onClose,
       children,
       className,
-      HeaderComponent,
-      closeButtonPosition,
-      mountEntry,
-      ...props
+      show,
     } = this.props
-    const cnRoot = 'mc-components-modal'
-
-    const classNames = cn(
-      `${cnRoot}__container`,
-      { [className]: Boolean(className) },
-    )
-    const cnClose = cn(
-      `${cnRoot}__close-icon`,
-      `${cnRoot}__close-icon--${closeButtonPosition}`,
-    )
 
-    if (!isOpen) {
-      return false
+    if (!show) {
+      return null
     }
 
-    return (
-      <ModalPortal mountEntry={mountEntry}>
-        { /outside/.test(closeButtonPosition) &&
-          <Close
-            className={cnClose}
-            onClick={onClose}
-          /> }
-        <ClickOutside
-          divRef={this.container}
-          onClickOutside={this.onClickOutside}
+    const classes = cn({
+      [className]: className,
+      'mc-modal': true,
```

Comment:
Can you use another classname?

We had a lot of issues with this classname because there are other modal implementations in masterclass with the same classname.

Reply:
Part of integrating this will be replacing all other modal implementations.

Reply:
I agree, we should use a different class name. What if we end up changing the structure of the react component and update the styles to match? If we use this class name we're locked into supporting and testing the old modals every time this gets updated